api/custom-map-ui/geoname-search-ui.html

```diff
@@ -0,0 +1,452 @@
+    <template id="geometrycollection_temp">
+        <map-geometrycollection id="geometrycollection"></map-geometrycollection>
+    </template>
+
+    <!-- Templates to add features to geometry -->
+    <template id ="linestring">
+        <map-linestring>
+            <map-coordinates id="coords"></map-coordinates>
+        </map-linestring>
+    </template>
+
+    <template id="polygon">
+        <map-polygon id="poly">
+        </map-polygon>
+    </template>
+
+    <template id="point_temp">
+        <map-point id="point">
+        </map-point>
+    </template>
+
+    <!-- Template to add coordinates to features -->
+    <template id="coords_temp">
+        <map-coordinates id="coords"></map-coordinates>
+    </template>
+    <!-- ############################################### TEMPLATES ############################################### -->
+    
+    <script>
+        // Handling pressing Enter on focused search results divs
+        function handleEnter(e) {
+            var keycode = (e.keyCode ? e.keyCode : e.which);
+            if (keycode == '13') {
+                document.activeElement.click();
+            }
+        }
+
+        // Handling enter on the input search box, the search button is clicked
+        var input = document.getElementById("query");
+        input.addEventListener("keyup", function(event) {
+            if (event.keyCode === 13) {
+                event.preventDefault();
+                document.getElementById("searchBtn").click();
+            }
+        });
+
+        // Select the mapml-viewer
+        let mapp = document.querySelector("#map");
+
+        // Will store the json search results
+        var search_results = ""; 
+
+        // Computes the zoom extent based on bounding box's decimal degrees
+        // getZoomExtent: listof(Num, Num, Num, Num) -> Int(3-16)
+        function getZoomExtent (bbox) {
+            var x = [bbox[0], bbox[2]];
+            var y = [bbox[1], bbox[3]];
+            var difference = Math.max(Math.abs(x[0]-x[1]), Math.abs(y[0]-y[1])).toPrecision(2);
+
+            // choosing zoom level, based on the decimal degree extent
+            if (difference >=30) {
+                return 3;
+            } else if (difference >=25) {
+                return 4;
+            } else if (difference >= 10) {
+                return 5;
+            } else if (difference >= 5) {
+                return 6;
+            } else if (difference >= 3) {
+                return 7;
+            } else if (difference >= 2) {
+                return 8;
+            } else if (difference >= 1) {
+                return 9;
+            } else if (difference >= 0.7) {
+                return 10;
+            } else if (difference >= 0.5) {
+                return 11;
+            } else if (difference >= 0.2) {
+                return 12;
+            } else if (difference >= 0.1) {
+                return 13;
+            } else if (difference == 0.04) { // Point feature
```

Comment:
In JavaScript I think it's recommended to use `===` rather than `==` for comparing, `==` does type conversion before comparing which although may not be an issue in this case is just something to consider.

Same goes for most other parts of the code where `==` is used rather than `===`.